be/src/common/config.h

```diff
@@ -758,6 +757,8 @@ CONF_Int32(connector_io_tasks_adjust_interval_ms, "50");
 CONF_Int32(connector_io_tasks_adjust_step, "1");
 CONF_Int32(connector_io_tasks_adjust_smooth, "4");
 CONF_Int32(connector_io_tasks_slow_io_latency_ms, "50");
+CONF_mDouble(scan_use_query_mem_ratio, "0.25");
+CONF_Double(connector_scan_use_query_mem_ratio, "0.3");
```

Comment:
why only 0.3?

Reply:
I am doing local testing, importing data into the lineorder_flat table, with 20 million records imported at a time.

> insert into lineorder_flat_temp select * from emr_test.ssb_100g_zhangyan_parquet_lz4.lineorder_flat limit $limitoffset $i;

The be.conf configuration has mem_limit set to 16GB. Memory allocation is roughly as follows:

Process limit = mem_limit * 0.9, which is the available upper limit of memory for the entire BE.
- Query limit = Process limit * 0.8
- Load limit = Process limit * 0.3
- In addition, there are other memory overheads such as compaction and column pool.

<img width="1767" alt="Pasted Graphic 1" src="https://user-images.githubusercontent.com/1081215/233696429-81fe1232-5ef0-447b-ab4a-a9a83218a328.png">

---

If we set it to 0.75, the scan limit proportion will occupy 0.75 * 0.8 = 0.6, which, when combined with the load limit and other overheads, can easily reach the upper limit.

Setting it to 0.5 sometimes still results in a few occurrences of out-of-memory (OOM) errors. This may be due to inaccurate estimation of memory usage by the chunk source.

Currently, setting it to 0.3 seems to be working fine. Apart from the chunk source, we also need to allocate a portion of memory to the chunk buffer.